systemd/systemd.go

```diff
@@ -203,54 +155,51 @@ func (*systemd) LogReader(serviceNames []string, n string, follow bool) (io.Read
 
 var statusregex = regexp.MustCompile(`(?m)^(?:(.*?)=(.*))?$`)
 
-func (s *systemd) Status(serviceName string) (string, error) {
-	status, err := s.ServiceStatus(serviceName)
-	if err != nil {
-		return "", err
-	}
-
-	return status.StatusString(), nil
-}
-
-// A ServiceStatus holds structured service status information.
-type ServiceStatus struct {
-	ServiceFileName string `json:"service-file-name"`
-	LoadState       string `json:"load-state"`
-	ActiveState     string `json:"active-state"`
-	SubState        string `json:"sub-state"`
-	UnitFileState   string `json:"unit-file-state"`
-}
-
-func (status *ServiceStatus) StatusString() string {
-	return fmt.Sprintf("%s; %s; %s (%s)", status.UnitFileState, status.LoadState, status.ActiveState, status.SubState)
-}
-
-func (s *systemd) ServiceStatus(serviceName string) (*ServiceStatus, error) {
-	bs, err := SystemctlCmd("show", "--property=Id,LoadState,ActiveState,SubState,UnitFileState", serviceName)
+func (s *systemd) Status(serviceNames ...string) ([]*client.ServiceInfo, error) {
+	cmd := make([]string, len(serviceNames)+2)
+	copy(cmd[2:], serviceNames)
+	cmd[0] = "show"
+	cmd[1] = "--property=Id,ActiveState,UnitFileState"
+	bs, err := SystemctlCmd(cmd...)
 	if err != nil {
 		return nil, err
 	}
 
-	status := &ServiceStatus{ServiceFileName: serviceName}
+	sts := make([]*client.ServiceInfo, 0, len(serviceNames))
+	cur := &client.ServiceInfo{}
 
 	for _, bs := range statusregex.FindAllSubmatch(bs, -1) {
-		if len(bs[0]) > 0 {
-			k := string(bs[1])
-			v := string(bs[2])
-			switch k {
-			case "LoadState":
-				status.LoadState = v
-			case "ActiveState":
-				status.ActiveState = v
-			case "SubState":
-				status.SubState = v
-			case "UnitFileState":
-				status.UnitFileState = v
+		if len(bs[0]) == 0 {
+			// systemctl separates data pertaining to particular services by an empty line
+			sts = append(sts, cur)
```

Comment:
should we be paranoid and be unhappy if cur.ServiceFileName or cur.Type are empty here?